multicorn_das/__init__.py

```diff
@@ -623,209 +781,212 @@ def raw_value_to_python(v):
+    if isinstance(v, Decimal):
+        return Value(decimal=ValueDecimal(v=str(v)))
+    if isinstance(v, time):
+        return Value(time=ValueTime(
+            hour=v.hour,
+            minute=v.minute,
+            second=v.second,
+            nano=v.microsecond * 1000
+        ))
+    if isinstance(v, datetime):
+        return Value(timestamp=ValueTimestamp(
+            year=v.year,
+            month=v.month,
+            day=v.day,
+            hour=v.hour,
+            minute=v.minute,
+            second=v.second,
+            nano=v.microsecond * 1000
+        ))
+    if isinstance(v, date):
+        # Note that datetime is also a date, so we check datetime first above
+        return Value(date=ValueDate(
+            year=v.year,
+            month=v.month,
+            day=v.day
+        ))
+
+    if isinstance(v, dict):
+        # Possibly an interval-like or record-like
+        # This is up to you how you interpret dict -> interval
+        # Example interval dict:
+        #   {'years':..., 'months':..., 'days':..., 'hours':..., 'minutes':..., 'seconds':..., 'micros':...}
+        # If that matches, build a ValueInterval. Otherwise treat as record.
+        if all(k in v for k in ('years', 'months', 'days', 'hours', 'minutes', 'seconds', 'micros')):
+            return Value(interval=ValueInterval(
+                years=v['years'],
+                months=v['months'],
+                days=v['days'],
+                hours=v['hours'],
+                minutes=v['minutes'],
+                seconds=v['seconds'],
+                micros=v['micros']
+            ))
+        else:
+            # treat as record
+            atts = []
+            for name, value in v.items():
+                atts.append(ValueRecordAttr(name=name, value=python_value_to_das(value)))
+            return Value(record=ValueRecord(atts=atts))
+
+    if isinstance(v, list):
+        inner_values = []
+        for value in v:
+            inner_values.append(python_value_to_das(value))
+        return Value(list=ValueList(values=inner_values))
+
+    log_to_postgres(f'Unsupported Python value: {v}', WARNING)
+    return None
 
 
 def operator_to_grpc_operator(operator):
-    if operator == '=':
+    """
+    Map a Python/SQL operator string to the gRPC Operator enum.
+    Returns None if not recognized.
+    """
+    op = operator.upper()
+    if op in ['=', '==']:
         return Operator.EQUALS
-    elif operator in ['<>', '!=']:
+    elif op in ['<>', '!=']:
         return Operator.NOT_EQUALS
-    elif operator == '<':
+    elif op == '<':
         return Operator.LESS_THAN
-    elif operator == '<=':
+    elif op == '<=':
         return Operator.LESS_THAN_OR_EQUAL
-    elif operator == '>':
+    elif op == '>':
         return Operator.GREATER_THAN
-    elif operator == '>=':
+    elif op == '>=':
         return Operator.GREATER_THAN_OR_EQUAL
-    elif operator == '~~':
+    elif op in ['~~', 'LIKE']:
```

Comment:
Why the string version is there now?

Reply:
For future PostgreSQL compatibility. BTW, it should be op.upper()

Reply:
It's fixed (`op = operator.upper()` at the top).